sw/airborne/modules/nav/nav_rotorcraft_hybrid.c

```diff
@@ -248,28 +274,30 @@ static void nav_hybrid_circle(struct EnuCoor_f *wp_center, float radius)
     VECT2_COPY(nav.target, *wp_center);
   }
   // compute desired speed
+  float radius_diff = fabsf(float_vect2_norm(&pos_diff) - abs_radius);
+  if (radius_diff > NAV_HYBRID_NAV_CIRCLE_DIST) {
+    // far from circle, speed proportional to diff
+    desired_speed = radius_diff * nav_hybrid_pos_gain;
+    nav_max_acceleration_sp = nav_hybrid_max_acceleration / 2.0;
+  } else {
+    // close to circle, speed function of radius for a feasible turn
+    // 0.8 * MAX_BANK gives some margins for the turns
+    desired_speed = sqrtf(PPRZ_ISA_GRAVITY * abs_radius * tanf(0.8f * nav_hybrid_max_bank));
+    nav_max_acceleration_sp = nav_hybrid_max_acceleration ;
+  }
   if (force_forward) {
     desired_speed = nav_max_speed;
-  } else {
-    float radius_diff = fabsf(float_vect2_norm(&pos_diff) - abs_radius);
-    if (radius_diff > NAV_HYBRID_NAV_CIRCLE_DIST) {
-      // far from circle, speed proportional to diff
-      desired_speed = radius_diff * nav_hybrid_pos_gain;
-    } else {
-      // close to circle, speed function of radius for a feasible turn
-      // 0.8 * MAX_BANK gives some margins for the turns
-      desired_speed = sqrtf(PPRZ_ISA_GRAVITY * abs_radius * tanf(0.8f * nav_hybrid_max_bank));
-    }
-    Bound(desired_speed, 0.0f, nav_max_speed);
+    nav_max_acceleration_sp = nav_hybrid_max_acceleration ;
   }
+  Bound(desired_speed, 0.0f, nav_max_speed);
   // compute speed vector from target position
   struct FloatVect2 speed_unit;
   VECT2_DIFF(speed_unit, nav.target, *stateGetPositionEnu_f());
   float_vect2_normalize(&speed_unit);
   VECT2_SMUL(nav.speed, speed_unit, desired_speed);
 
   nav_rotorcraft_base.circle.center = *wp_center;
-  nav_rotorcraft_base.circle.radius = radius;
+  nav_rotorcraft_base.circle.radius = sign_radius * abs_radius;
```

Comment:
Why is this better than just radius?

Reply:
Because abs_radius is the bounded variable.